app/uk/gov/hmrc/advancevaluationrulings/connectors/DmsSubmissionConnector.scala

```diff
@@ -64,11 +63,26 @@ class DmsSubmissionConnector @Inject() (
   private def fileName(attachment: DmsAttachment, name: String): String =
     (attachment.privacy, attachment.isLetterOfAuthority) match {
 
-      case (_, true)                 => "Letter_of_authority." + FilenameUtils.getExtension(name)
+      case (_, true)                 => "Letter_of_authority." + getExtension(name)
       case (Privacy.Confidential, _) => s"CONFIDENTIAL_$name"
       case _                         => name
     }
 
+  private def getExtension(filename: String): String =
+    if (filename == null || filename.trim.isEmpty) {
+      logger.error("[DmsSubmissionConnector][getExtension] Invaild file extension")
+      ""
+      // "invalid" // Handle null or empty filename
+    } else {
+      val dotIndex = filename.lastIndexOf('.')
+      if (dotIndex >= 0 && dotIndex < filename.length - 1) {
+        filename.substring(dotIndex + 1).toLowerCase
+      } else {
+        logger.error("[DmsSubmissionConnector][getExtension] Invaild file extension")
+        ""
+      }
+    }
```

Comment:
```suggestion
private def getExtension(filename: String): String = {
  Option(filename)
    .filter(_.trim.nonEmpty)
    .map(_.lastIndexOf('.'))
    .filter(dotIndex => dotIndex >= 0 && dotIndex < filename.length - 1)
    .map(dotIndex => filename.substring(dotIndex + 1).toLowerCase)
    .getOrElse {
      logger.error("[DmsSubmissionConnector][getExtension] Invalid file extension")
      ""
    }
}
```

Reply:
Option is asafer way of handling null and this way we can remove the nested if statement. It also alows for just the one logging